server/src/main/java/io/druid/client/CoordinatorServerView.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.client;
+
+import com.google.api.client.util.Sets;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Ordering;
+import com.google.inject.Inject;
+import com.metamx.common.logger.Logger;
+import io.druid.concurrent.Execs;
+import io.druid.query.DataSource;
+import io.druid.server.coordination.DruidServerMetadata;
+import io.druid.timeline.DataSegment;
+import io.druid.timeline.VersionedIntervalTimeline;
+import io.druid.timeline.partition.PartitionChunk;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * ServerView of coordinator for the state of segments being loaded in the cluster.
+ */
+public class CoordinatorServerView implements TimelineInventoryView
```

Comment:
Why does this implement `TimelineInventoryView`?  I might not have read the code fully yet, but it seems like it would only expose TimelineInventoryView if it is hoping to expose things based on overshadowing (i.e. the "currently active" segments).  However, in my view, the coordinator would just be exposing information about the cluster, not necessarily only the currently active segments.

Rather, it seems like this should be exposing just a server-based view of all segments that exist and maybe also a segment-based view of all servers that have said segment.